frontera/contrib/backends/sqlalchemy/__init__.py

```diff
@@ -161,9 +166,15 @@ def _create_page(self, obj):
         db_page.fingerprint = obj.meta['fingerprint']
         db_page.state = Page.State.NOT_CRAWLED
         db_page.url = obj.url
-        db_page.depth = 0
         db_page.created_at = datetime.datetime.utcnow()
         db_page.meta = obj.meta
+
+        if not isinstance(obj, frontera_response):
```

Comment:
@RajatGoyal could you try to use there canonical solver? (let me know if you need help) and then if your request/response causes creation of new page you should reset depth. It's independent on the ``obj`` type.